### PR TITLE
[MOS-943] Remove old factory reset mechanism

### PR DESCRIPTION
**Pull request by Lefucjusz**
Removed old factory reset mechanism that is
no longer needed after UDM.
Fixed accessing previously moved variable.
Minor cleanups.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

File: module-services/service-db/ServiceDBCommon.cpp [hunk 1]
```diff
@@ -25,25 +28,26 @@ db::Interface *ServiceDBCommon::getInterface(db::Interface::Name interface)
...
-        db::Interface *interface = getInterface(msg->getInterface());
+
+        const auto interface = getInterface(msg->getInterface());
         assert(interface != nullptr);
-        auto query     = msg->getQuery();
-        auto queryType = query->type;
-        auto result    = interface->runQuery(std::move(query));
-        std::optional<uint32_t> id;
+
+        const std::shared_ptr<db::Query> query(std::move(msg->getQuery()));
```
Comment (swichu91) [inline module-services/service-db/ServiceDBCommon.cpp:40]: auto?

File: products/PurePhone/services/db/ServiceDB.cpp [hunk 2]
```diff
@@ -85,7 +86,7 @@ sys::MessagePointer ServiceDB::DataReceivedHandler(sys::DataMessage *msgl, sys::
 
     case MessageType::DBContactAdd: {
         auto time             = utils::time::Scoped("DBContactAdd");
-        DBContactMessage *msg = reinterpret_cast<DBContactMessage *>(msgl);
+        auto msg              = dynamic_cast<DBContactMessage *>(msgl);
```
Comment (swichu91) [inline products/PurePhone/services/db/ServiceDB.cpp:89]: I believe in that case you can safely use `static_cast` or at least(if you really want to precise), check the result of `dynamic_cast`. This also applies to the rest of code below ;)

**Review by swichu91 — APPROVED** [review]
Looks good! Please check suggestions:)

File: products/PurePhone/services/db/ServiceDB.cpp (hunk 2 above)
Comment (Lefucjusz) [reply products/PurePhone/services/db/ServiceDB.cpp]: Changed to `static_cast` :)

**Review by awulkiew — APPROVED** [review]

**Pull request merged**